teamcity: failed tests on release-2.0: acceptance/TestJSONBUpgrade

The following tests appear to have failed:

[#637200](https://teamcity.cockroachdb.com/viewLog.html?buildId=637200):


```
--- FAIL: acceptance/TestJSONBUpgrade (1.680s)
test_log_scope.go:81: test logs captured to: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/logTestJSONBUpgrade456234567
test_log_scope.go:62: use -show-logs to present logs inline
--- FAIL: acceptance/TestJSONBUpgrade: TestJSONBUpgrade/runMode=local (1.680s)
util_cluster.go:214: dial tcp 127.0.0.1:5432: connect: connection refused
------- Stdout: -------
CockroachDB node starting at 2018-05-02 17:07:48.072012516 +0000 UTC (took 0.5s)
build:      CCL v1.1.8 @ 2018/04/23 17:25:48 (go1.8.3)
admin:      http://127.0.0.1:46257
sql:        postgresql://root@127.0.0.1:37617?application_name=cockroach&sslmode=disable
logs:       /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestJSONBUpgrade/runMode=local/1
store[0]:   path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster214097658/1
status:     initialized new cluster
clusterID:  2e260a5e-9df5-4e3b-9d91-0f80a2c8ce7e
nodeID:     1
test logs left over in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/logTestJSONBUpgrade456234567
--- FAIL: acceptance/TestJSONBUpgrade (1.680s)
test_log_scope.go:81: test logs captured to: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/logTestJSONBUpgrade456234567
test_log_scope.go:62: use -show-logs to present logs inline
--- FAIL: acceptance/TestJSONBUpgrade: TestJSONBUpgrade/runMode=local (1.680s)
util_cluster.go:214: dial tcp 127.0.0.1:5432: connect: connection refused
------- Stdout: -------
CockroachDB node starting at 2018-05-02 17:07:48.072012516 +0000 UTC (took 0.5s)
build:      CCL v1.1.8 @ 2018/04/23 17:25:48 (go1.8.3)
admin:      http://127.0.0.1:46257
sql:        postgresql://root@127.0.0.1:37617?application_name=cockroach&sslmode=disable
logs:       /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestJSONBUpgrade/runMode=local/1
store[0]:   path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster214097658/1
status:     initialized new cluster
clusterID:  2e260a5e-9df5-4e3b-9d91-0f80a2c8ce7e
nodeID:     1
test logs left over in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/logTestJSONBUpgrade456234567

```

Please assign, take a look and update the issue accordingly.

--- a/pkg/acceptance/localcluster/cluster.go
+++ b/pkg/acceptance/localcluster/cluster.go
@@ -89,6 +89,7 @@ type ClusterConfig struct {
 	PerNodeCfg  map[int]NodeConfig // optional map of nodeIndex -> configuration
 	DB          string             // database to configure DB connection for
 	NumWorkers  int                // SetMaxOpenConns to use for DB connection
+	NoWait      bool               // if set, return from Start before cluster ready
 }
 
 // NodeConfig is a configuration for a node in a Cluster. Options with the zero
@@ -199,7 +200,7 @@ func (c *Cluster) Start(ctx context.Context) {
 		}
 	}
 
-	if c.Cfg.NumNodes > 1 {
+	if !c.Cfg.NoWait {
 		for i := range chs {
 			if err := <-chs[i]; err != nil {
 				log.Fatalf(ctx, "node %d: %s", i+1, err)
@@ -209,7 +210,7 @@ func (c *Cluster) Start(ctx context.Context) {
 
 	log.Infof(context.Background(), "started %.3fs", timeutil.Since(c.started).Seconds())
 
-	if c.Cfg.NumNodes > 1 {
+	if c.Cfg.NumNodes > 1 || !c.Cfg.NoWait {
 		c.waitForFullReplication()
 	} else {
 		// NB: This is useful for TestRapidRestarts.

--- a/pkg/acceptance/util_cluster.go
+++ b/pkg/acceptance/util_cluster.go
@@ -135,6 +135,7 @@ func StartCluster(ctx context.Context, t *testing.T, cfg cluster.TestConfig) (c 
 			LogDir:     logDir,
 			NumNodes:   len(cfg.Nodes),
 			PerNodeCfg: perNodeCfg,
+			NoWait:     cfg.NoWait,
 		}
 
 		l := localcluster.New(clusterCfg)

